caddy/caddy_test.go

```diff
@@ -305,3 +308,153 @@ func TestPHPServerDirectiveDisableFileServer(t *testing.T) {
 	tester.AssertGetResponse("http://localhost:9080", http.StatusOK, "I am by birth a Genevese (i not set)")
 	tester.AssertGetResponse("http://localhost:9080/hello.txt", http.StatusNotFound, "Not found")
 }
+
+func TestMetrics(t *testing.T) {
+	var wg sync.WaitGroup
+	tester := caddytest.NewTester(t)
+	tester.InitServer(`
+	{
+		skip_install_trust
+		admin localhost:2999
+		http_port 9080
+		https_port 9443
+
+		frankenphp
+	}
+
+	localhost:9080 {
+		route {
+			php {
+				root ../testdata
+			}
+		}
+	}
+	`, "caddyfile")
+
+	// Make some requests
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func(i int) {
+			tester.AssertGetResponse(fmt.Sprintf("http://localhost:9080/index.php?i=%d", i), http.StatusOK, fmt.Sprintf("I am by birth a Genevese (%d)", i))
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+
+	// Fetch metrics
+	resp, err := http.Get("http://localhost:2999/metrics")
+	if err != nil {
+		t.Fatalf("failed to fetch metrics: %v", err)
+	}
+	defer resp.Body.Close()
+
+	// Read and parse metrics
+	metrics := new(bytes.Buffer)
+	_, err = metrics.ReadFrom(resp.Body)
+	if err != nil {
+		t.Fatalf("failed to read metrics: %v", err)
+	}
+
+	cpus := fmt.Sprintf("%d", runtime.GOMAXPROCS(0))
```

Comment:
IIRC, by default, the number of threads is 2 times the number of available CPU.

Maybe could we find a better default value btw.

Reply:
How would you feel exposing this?

`frankenphp.MaxThreads()` or something.

That would make this test less brittle and make tuning it much easier.

Reply:
yes, why not! Or even just an exported field.

Reply:
And done! I also increased the max threads to `cpu_count*2` as only workers were considered in that calculation. I added a test for it as well.